### ISSUE TITLE
How to use the location 2-4 of reads at  5' ? 

### ISSUE THREAD

**Issue by biofer** [OpenGene/fastp#417]
I want to remove the begin 1 bp and use the 2-4 bp as UMI ,  did `fastp` support this?

```
fastp ... \
      --trim_front1 1  --trim_front1 1 \
      -U --umi_loc per_read --umi_prefix UMI --umi_len 3 --umi_skip 3 
```
and it use the 1-3bp as UMI。

**Comment by jan-glx** [comment]
No. As [described in the readme](https://github.com/OpenGene/fastp#global-trimming) UMIs are extracted before global trimming. But `fastp` is nice, just run it twice:
```
fastp ... \
  --trim_front1 1  --trim_front1 1 \
  --stdout | fastp --stdin \
  -U --umi_loc per_read --umi_prefix UMI --umi_len 3 --umi_skip 3 
```
Even in case `fastp` trimmed before extracting UMIs and restarted counting bases after global trimming, this makes your intentions explicit and it easier for others to follow your code. Note, that `fastp` will most likely never be the bottleneck of your analysis. Alternatively you could just keep the first base as part of the UMI and, if necessary, remove it later.

**Comment by jan-glx** [comment]
See also #447
